pkg/policy/selectorcache.go

```diff
@@ -446,12 +492,15 @@ func (sc *SelectorCache) AddFQDNSelector(user CachedSelectionUser, fqdnSelec api
 			cachedSelections: make(map[identity.NumericIdentity]struct{}),
 		},
 		selector: fqdnSelec,
+		user:     sc.pop,
 	}
 
 	// Add the initial user
 	newFQDNSel.users[user] = struct{}{}
 	newFQDNSel.updateSelections()
```

Comment:
This should guarantee that the `selections` member is always initialized, so `GetSelections()` should not require the `nil` check?